### PR TITLE
Show FG_NHG CLI Commands Added

### PR DESCRIPTION
**Pull request by kktheballer**
1. show fg-nhg-hash-view
   (shows the current hash bucket view of fg nhg)

2. show fg-nhg-active-hops
   (shows which set of next-hops are active)

**- What I did**

**- How I did it**

**- How to verify it**

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

### CODE REVIEW

**Comment by lguohan** [comment]
please add unit test and also check the coverage report generated.

**Comment by lguohan** [comment]
please address lgtm warnings

File: show/main.py [hunk 1]
```diff
@@ -1917,6 +1917,210 @@ def mmu():
     click.echo(proc.stdout.read())
 
 
+#
```
Comment (lguohan) [inline show/main.py:1920]: since this is a new group, please separate into a different file. check show/vlan.py for example.

File: show/main.py [hunk 2]
```diff
@@ -1917,6 +1917,210 @@ def mmu():
...
+# 'fg nhg active-hops' command ("show fg-nhg-active-hops")
+#
+@cli.command('fg-nhg-active-hops')
+@click.argument('nhg', required=True)
+def fg_nhg_active_hops(nhg):
+    state_db = SonicV2Connector(host='127.0.0.1')
```
Comment (lguohan) [inline show/main.py:1926]: please use pass_db decorator.

File: show/main.py [hunk 3]
```diff
@@ -1917,6 +1917,210 @@ def mmu():
...
+    try:
+        state_db.connect(state_db.STATE_DB, False)  # Make one attempt only STATE_DB
+    except:
+        click.echo("STATE_DB Table does not exist!")
```
Comment (lguohan) [inline show/main.py:1930]: use ctx.fail()

File: show/main.py [hunk 4]
```diff
@@ -1916,6 +1916,217 @@ def mmu():
...
+    table = []
+    key_avail = {"ipv4":None, "ipv6": None}
+
+    for tk in table_keys:
+	if "." in tk:
+		key_avail['ipv4'] = tk
+	elif ":" in tk:
+		key_avail['ipv6'] = tk
+
+    if nhg == "fgnhg_v4":
```
[the remark this answers is not in the document]
Comment (kktheballer) [reply show/main.py]: addressed

File: show/main.py [hunk 5]
```diff
@@ -1916,6 +1916,217 @@ def mmu():
...
+	if key_avail["ipv4"] is not None:
+		t_dict = state_db.get_all(state_db.STATE_DB, key_avail["ipv4"])
+
+	else:
+		click.echo("IPV4 nhg does not exist!")
+		exit()
+
+    if nhg == "fgnhg_v6":
```
[the remark this answers is not in the document]
Comment (kktheballer) [reply show/main.py]: addressed

File: show/main.py (hunk 2 above)
Comment (kktheballer) [reply show/main.py]: For some reason, when locally testing using pass_db decorator I wasn't able to fetch my config_db table using "db.cfgdb.get_table()".

Only when using SonicV2Connector was I able to fetch my config_db table. Any suggestions?

**Comment by kktheballer** [comment]
retest default please

**Comment by kktheballer** [comment]
retest this please

**Comment by kktheballer** [comment]
retest this please

**Comment by kktheballer** [comment]
retest this please

**Comment by kktheballer** [comment]
retest this please

File: show/ecmp.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
+import click
+import ipaddress
+from tabulate import tabulate
+from swsssdk import ConfigDBConnector
+from swsssdk import SonicV2Connector
+import utilities_common.cli as clicommon
+from collections import OrderedDict
+
+@click.group(cls=clicommon.AliasedGroup)
+def ecmp():
```
Comment (lguohan) [inline show/ecmp.py:10]: i feel the name should be fgnhg, whic is it ecmp?

**Comment by kktheballer** [comment]
retest this please -vv

**Comment by kktheballer** [comment]
retest this please

**Comment by kktheballer** [comment]
retest this please

**Comment by lguohan** [comment]
can you paste the coverage report results?

**Comment by kktheballer** [comment]
image

File: tests/fgnhg_test.py [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+| FG_NHG_PREFIX           | Active Next Hops   |
++=========================+====================+
+| NHG_PREFIX: fc:05::/128 | 200:200:200:200::4 |
+|                         | 200:200:200:200::5 |
++-------------------------+--------------------+
+"""
+
+
+
+class TestECMP(object):
```
Comment (lguohan) [inline tests/fgnhg_test.py:197]: just minor comments, better to be rename to TestFineGrainedNexthopGroup

**Review by lguohan — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Comment by kktheballer** [comment]
retest this please

**Pull request merged**